substrate/frame/revive/src/lib.rs

```diff
@@ -1460,20 +1460,22 @@ impl<T: Config> Pallet<T> {
 		let mut call_info = create_call::<T>(tx, None)
 			.map_err(|err| EthTransactError::Message(format!("Invalid call: {err:?}")))?;
 
-		// emulate transaction behavior
+		// the dry-run might leave out certain fields
+		// in those cases we skip the check that the caller has enough balance
+		// to pay for the fees
 		let fees = call_info.tx_fee.saturating_add(call_info.storage_deposit);
-		match (gas, from, value.is_zero()) {
-			(Some(_), Some(from), _) | (_, Some(from), false) => {
-				let balance = Self::evm_balance(&from).saturating_add(value);
-				if balance < Pallet::<T>::convert_native_to_evm(fees) {
-					return Err(EthTransactError::Message(format!(
-								"insufficient funds for gas * price + value: address {from:?} have {balance:?} (supplied gas {gas:?})",
-					)));
-				}
-			},
-			_ => {},
+		if let Some(from) = &from {
+			let fees = if gas.is_some() { fees } else { Zero::zero() };
+			let balance = Self::evm_balance(from).saturating_sub(value);
+			if balance < Pallet::<T>::convert_native_to_evm(fees) {
+				return Err(EthTransactError::Message(format!(
+					"insufficient funds for gas * price + value ({fees:?}): address {from:?} have {balance:?} (supplied gas {gas:?})",
+				)));
+			}
```

Comment:
in my tests against Geth, the error is also triggered if some value is passed and a user is specified

Reply:
So you think the old behavior was correct? i.e we should perform the check even if `gas` is `None`?

Reply:
need to look again looks like evm tests are green though

Reply:
The new code still does check of you have enough to cover the `value`. It just skips the check for the txfee if no `gas` is specified. Because that value will the block gas limit and `from` might not have enough to pay for all of that.